frontend/src/components/ViewTrips/trip.js

```diff
@@ -46,8 +46,28 @@ const Trip = (props) => {
   const destination = props.tripData[DB.TRIPS_DESTINATION];
   const startDateTimestamp = props.tripData[DB.TRIPS_START_DATE];
   const endDateTimestamp = props.tripData[DB.TRIPS_END_DATE];
-  const collaboratorEmailsStr =
-      getCollaboratorEmails(props.tripData[DB.TRIPS_COLLABORATORS]);
+  const collaboratorUidArr = props.tripData[DB.TRIPS_COLLABORATORS];
+  const [collaboratorEmailsStr, setCollaboratorEmailsStr] = useState('');
+
+  useEffect(() => {
+    // Only set state collaboratorEmailsStr if component is mounted. This is
+    // a precautionary to mitigate warnings that occur when setting state on
+    // a component that has already unmounted. See more here
+    // https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component.
+    let componentStillMounted = true;
+
+    async function fetchCollaboratorEmails() {
+      let collaboratorEmailArr =
+          await authUtils.getUserEmailArrFromUserUidArr(collaboratorUidArr);
+      collaboratorEmailArr = moveCurUserEmailToFront(collaboratorEmailArr);
+      if (componentStillMounted) {
+        setCollaboratorEmailsStr(collaboratorEmailArr.join(', '));
+      }
+    }
+
+    fetchCollaboratorEmails();
+    return () => { componentStillMounted = false; };
```

Comment:
Same as the comment I left on #113.

Reply:
Merged up the PR chain from #113. Fixed now.